views-soy/src/main/java/io/micronaut/views/soy/SoySauceViewsRenderer.java

```diff
@@ -132,62 +205,51 @@ public String getTemplateName() {
         return null;
       }
     });
-    renderer.setData(context);
-    if (injectNonce && request != null) {
-      Optional<Object> nonceObj = request.getAttribute(CspFilter.NONCE_PROPERTY);
-      if (nonceObj.isPresent()) {
-        String nonceValue = ((String) nonceObj.get());
-        ijOverlay.put(INJECTED_NONCE_PROPERTY, nonceValue);
-      }
-    }
-    renderer.setIj(ijOverlay);
+    renderer.setData(context.getProperties());
+    renderer.setIj(context.getInjectedProperties());
 
-    if (this.soyMicronautConfiguration.isRenamingEnabled() && this.namingMapProvider != null) {
-      SoyCssRenamingMap cssMap = this.namingMapProvider.cssRenamingMap();
-      SoyIdRenamingMap idMap = this.namingMapProvider.idRenamingMap();
-      if (cssMap != null) {
-        renderer.setCssRenamingMap(cssMap);
-      }
-      if (idMap != null) {
-        renderer.setXidRenamingMap(idMap);
+    if (this.soyMicronautConfiguration.isRenamingEnabled()) {
+      SoyNamingMapProvider renamingProvider = (
+        context.overrideNamingMap().orElse(this.namingMapProvider));
+      if (renamingProvider != null) {
+        SoyCssRenamingMap cssMap = renamingProvider.cssRenamingMap();
+        SoyIdRenamingMap idMap = renamingProvider.idRenamingMap();
+        if (cssMap != null) {
+          renderer.setCssRenamingMap(cssMap);
+        }
+        if (idMap != null) {
+          renderer.setXidRenamingMap(idMap);
+        }
       }
     }
 
-    try {
-      final AppendableToWritable target = new AppendableToWritable();
-      SoySauce.WriteContinuation state;
-
-      state = renderer.renderHtml(target);
-
-      while (state.result().type() != RenderResult.Type.DONE) {
-        switch (state.result().type()) {
-          // If it's done, do nothing.
-          case DONE: break;
+    // prime the initial render
+    return Flowable.<ByteBuffer, SoyRender>generate(SoyRender::create, (buffer, emitter) -> {
```

Comment:
core of the new render flow happens here, where we start with `Flowable.<>generate` and proceed until we either switch out of execution, or finish rendering.

if we switch out of execution, the engine hands us a future, which we wait on reactively. once the future completes, we continue rendering until finished, or until we are limited again by a future, causing the process to repeat.

in the simplest case, a render context has no future values, and proceeds regularly in a de-facto synchronous manner, populating the response in one go and sending it back. in this case, there is no chunked response, it's just a regular HTTP cycle.

in the most complex case, a render context has multiple future values that have not yet resolved, while hitting the soft limit of the buffer in between. it renders until it must wait, then switches out. once the future completes, it switches back in, continues rendering, and let's say it hits the buffer soft limit. so, the buffer indicates back pressure, causing a chunked response (and the initial chunk), then continues processing, encounters the 2nd future, switches out. when the second future completes, it switches back in, completes the render, emits the final chunk, and finishes.

see caveats in PR description for chunked response switching.